activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -581,11 +581,23 @@ def load_additional_types(type_map, oids = nil) # :nodoc:
           end
         end
 
+        def prepared_statements?(sql)
+          sql =~ /\$\d+/i
```

Comment:
There was a recent cleanup in the code base to not use `=~` where appropriate, see 99cf7558000090668b137085bfe6bcc06c4571dc for changes in AR and 575dbeeefcaafeb566afc07cdd8b55603b698d9f for the rationale.